### PR TITLE
[quantization] use channel scales for w4a8 + misc fixes

### PR DESCRIPTION
**Pull request by czhu-cohere**
## Purpose
Load per-channel scales for w4a8. This can recover the quality drop from naively casting bf16 scales to fp8 on certain benchmarks (mmlu pro). Computationally, this is 'free' since the [previous implementation](https://github.com/vllm-project/vllm/pull/23198) used `torch.ones` as a placeholder.

The `fp8` group scales and `fp32` per-channel scales can be generated as a post-processing step after we have a w4a16 checkpoint with `bf16` scales. For testing we used an adhoc workflow to generate the checkpoint but will look at integrating that into `llm-compressor` as a next step:
1. Take bf16 scales from the w4a16 checkpoint
2. do `fp8_scales, fp32_chan_scales = quantfp8(bf16_scales)`
3. divide `fp8_scales` by 8 to avoid saturation when multiplied by int4
4. multiply `fp32_chan_scales` by 8 to compensate

Then pass the adjusted `fp8_scales/fp32_chan_scales` to the `w4a8` kernel.

## Test Plan
lm-eval (gsm8k, mmlu pro) compare to w4a16 and previous w4a8 (Cohere Command A)

also add an example model `czhu-cohere/TinyLlama-1.1B-Chat-v1.0-W4A8-e2e` and add corresponding test in `tests/quantization/test_compressed_tensors.py`
```
pytest tests/quantization/test_compressed_tensors.py::test_compressed_tensors.py::test_compressed_tensors_w4a8_fp8
```
## Test Result
```
Cohere Command A (111B)
|model        |gsm8k       |mmlu pro    |
|-------------|------------|------------|
|w4a16        |0.8483699773|0.6907413564|
|w4a8 (before)|0.8476118271|0.6760305851|
|w4a8 (after) |0.8514025777|0.6880817819|
```

```
pytest tests/quantization/test_compressed_tensors.py::test_compressed_tensors.py::test_compressed_tensors_w4a8_fp8
...
collected 1 item                                                                                                                

test_compressed_tensors.py::test_compressed_tensors_w4a8_fp8[args0] PASSED                                                [100%]

====================================================== 1 passed in 52.44s =======================================================
```
## (Optional) Documentation Update

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

File: vllm/model_executor/layers/quantization/kernels/mixed_precision/cutlass.py [hunk 1]
```diff
@@ -86,19 +90,15 @@ def transform_w_s(x):
...
+        self._transform_param(layer, "weight_chan_scale", lambda x: x)
 
     def apply_weights(self,
                       layer: torch.nn.Module,
                       x: torch.Tensor,
                       bias: Optional[torch.Tensor] = None) -> torch.Tensor:
-        assert bias is None, "bias not supported by CUTLASS W4A8"
         c = self.config
         w_q, w_s, _, _ = self._get_weight_params(layer)
+        w_ch_s = layer.weight_chan_scale
```
Comment (czhu-cohere) [inline vllm/model_executor/layers/quantization/kernels/mixed_precision/cutlass.py:101]: I feel this is better than modifying every place `self._get_weight_params` is called

Comment (LucasWilkinson) [reply vllm/model_executor/layers/quantization/kernels/mixed_precision/cutlass.py:101]: Agreed; we eventually need to refactor `MPLinearKernel` to get rid of `_get_weight_params` which seems possible with torch 2.8.0 where we may not need to re-register the params in `process_weights_after_loading` and can just map everything to consistent names in `MPLinearKernel`

**Review by LucasWilkinson — APPROVED** [review]
Nice work! Thanks for the contribution!

**Comment by dsikka** [comment]
@LucasWilkinson can you attach the `ready` label so we can run the full quant test
